tests/unit/utils/updateNickname.test.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  INTERNAL_SERVER_ERROR,
+  NAME_CHANGED,
+} from "../../../src/constants/responses";
+import { DISCORD_BASE_URL } from "../../../src/constants/urls";
+import JSONResponse from "../../../src/utils/JsonResponse";
+import { updateNickName } from "../../../src/utils/updateNickname";
+
+describe("Update nickname", () => {
+  const mockEnv = {
+    BOT_PUBLIC_KEY: "xyz",
+    DISCORD_GUILD_ID: "123",
+    DISCORD_TOKEN: "abc",
+  };
+
+  const mockData = { discordId: "12345678910111213", nickname: "jhon" };
+
+  test("updatenickname fetch is called with expected parameters", async () => {
+    const data = { nick: mockData.nickname };
```

Comment:
`nick` doesn't sound very convincing haha. We can name something better i hope 😂

Reply:
That cant be chosen by the developer discord has a pre defined key that needs to be send as body.
https://discord.com/developers/docs/resources/guild#modify-guild-member

Reply:
oh cool, that's hilarious.